livestream-compose-sample/src/main/java/io/getstream/livestream/compose/streams/CameraLiveStream.kt

```diff
@@ -0,0 +1,131 @@
+import io.getstream.chat.android.compose.handlers.SystemBackPressedHandler
+import io.getstream.chat.android.compose.ui.messages.list.MessageList
+import io.getstream.chat.android.compose.ui.theme.ChatTheme
+import io.getstream.chat.android.compose.viewmodel.messages.MessageComposerViewModel
+import io.getstream.chat.android.compose.viewmodel.messages.MessageListViewModel
+import io.getstream.livestream.compose.LiveStreamHeader
+import io.getstream.livestream.compose.LiveStreamMessage
+import io.getstream.livestream.compose.LiveStreamMessageList
+import io.getstream.livestream.compose.LivestreamComposer
+
+/**
+ * Shows a camera preview surface in a view component that relies on [MessageListViewModel]
+ * and [MessageComposerViewModel] to connect all the chat data handling operations.
+ *
+ * @param composerViewModel - [MessageComposerViewModel] for manging message input field
+ * @param listViewModel - [MessageListViewModel] The ViewModel that stores all the data and
+ * business logic required to show a list of messages. The user has to provide one in this case,
+ * as we require the channelId to start the operations.
+ * @param onBackPressed - Handler for when the user clicks back press
+ */
+@Composable
+fun CameraLiveStream(
+    composerViewModel: MessageComposerViewModel,
+    listViewModel: MessageListViewModel,
+    onBackPressed: () -> Unit
+) {
+    val lifecycleOwner = LocalLifecycleOwner.current
+    val context = LocalContext.current
+    val cameraProviderFuture = remember { ProcessCameraProvider.getInstance(context) }
+
+    LaunchedEffect(Unit) {
+        listViewModel.start()
+    }
+
+    ChatTheme {
+        Box(modifier = Modifier.fillMaxSize()) {
+            AndroidView(
+                factory = { ctx ->
+                    val preview = PreviewView(ctx)
+                    val executor = ContextCompat.getMainExecutor(ctx)
+                    cameraProviderFuture.addListener({
+                        val cameraProvider = cameraProviderFuture.get()
+                        bindPreview(
+                            lifecycleOwner,
+                            preview,
+                            cameraProvider
+                        )
+                    }, executor)
+                    preview
+                },
+                modifier = Modifier.fillMaxSize(),
+            )
+            Box(
+                modifier = Modifier
+                    .background(
+                        brush = Brush.verticalGradient(
+                            listOf(Color.Transparent, Color.Black),
+                            0f,
+                            1050f,
+                        )
+                    )
+                    .align(Alignment.BottomCenter)
+            ) {
+                Column {
+                    LiveStreamMessageList(listViewModel)
+                    LivestreamComposer(composerViewModel)
+                }
+            }
+
+            LiveStreamHeader {
```

Comment:
How does the header come after the `Box`?

Reply:
Can you help me with more details on this comment

Reply:
Realised the header is last probably because of the Z-axis/index. We can ignore this. But we should put a modifier to the header to align it to the top of the screen.

Reply:
sure , im getting used to the order (inside box , inside surface etc) 

checking with align modifier on the header

Reply:
Fixed